### PR TITLE
Feature/add spot allocation strategy options and mixed instances policy

### PR DESCRIPTION
**Pull request by matiasmarrapodi**

### CODE REVIEW

File: nodegroups.tf [hunk 1]
```diff
@@ -94,25 +93,9 @@ resource "aws_launch_template" "eks_node_groups" {
   }
 
   monitoring {
-    enabled = true
+    enabled = var.monitoring
```
Comment (ivours) [inline nodegroups.tf:96]: We can use a more descriptive variable name here, for example: `var.enable_detailed_monitoring`

File: nodegroups.tf [hunk 2]
```diff
@@ -138,24 +121,36 @@ resource "aws_autoscaling_group" "eks" {
   target_group_arns    = var.target_group_arns
   health_check_type    = var.health_check_type
 
-  launch_template {
-    id      = aws_launch_template.eks_node_groups[each.key].id
-    version = "$Latest"
+  mixed_instances_policy {
+    dynamic "instances_distribution" {
+      for_each =  each.value.spot_nodes_enabled == true ? ["do it"] : []
```
Comment (ivours) [inline nodegroups.tf:126]: The `== true` isn't needed here, since `each.value.spot_nodes_enabled` is already a boolean type.

File: nodegroups.tf [hunk 3]
```diff
@@ -138,24 +121,36 @@ resource "aws_autoscaling_group" "eks" {
...
+      }
+      }
+
+    launch_template {
+      launch_template_specification {
+        launch_template_id      = aws_launch_template.eks_node_groups[each.key].id
+        version = "$Latest"
+      }
+  }
   }
```
Comment (ivours) [inline nodegroups.tf:140]: Fix indentation

File: nodegroups.tf [hunk 4]
```diff
@@ -138,24 +121,36 @@ resource "aws_autoscaling_group" "eks" {
...
-    version = "$Latest"
+  mixed_instances_policy {
+    dynamic "instances_distribution" {
+      for_each =  each.value.spot_nodes_enabled == true ? ["do it"] : []
+      content {
+        spot_allocation_strategy = var.spot_allocation_strategy
+        on_demand_percentage_above_base_capacity = var.on_demand_percentage_above_base_capacity
+        spot_instance_pools = var.spot_instance_pools
+      }
+      }
```
Comment (ivours) [inline nodegroups.tf:132]: Fix indentation

File: variables.tf [hunk 5]
```diff
@@ -102,6 +116,11 @@ variable "enabled_cluster_log_types" {
     default = []
 }
 
+variable "monitoring" {
```
Comment (ivours) [inline variables.tf:119]: We can use a more descriptive variable name here, for example: var.enable_detailed_monitoring

**Review by ivours — CHANGES_REQUESTED** [review]

File: nodegroups.tf (hunk 1 above)
Comment (matiasmarrapodi) [reply nodegroups.tf]: Done

File: variables.tf (hunk 5 above)
Comment (matiasmarrapodi) [reply variables.tf]: Done

**Review by ivours — APPROVED** [review]

File: CHANGELOG.md [hunk 6]
```diff
@@ -6,6 +6,14 @@ The format is based on [Keep a Changelog](https://keepachangelog.com/en/1.0.0/),
 and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0.html).
 
 
+## [5.3.0] - 2024-07-29
+
+- Add mixed instances policy.
+- Add capacity across the Spot pools.
+- Add percentage split between on-demand and Spot instances.
+- Add number of Spot pools per availability zone to allocate capacity.
+- Add variable to choose whether to activate detail monitoring.
```
Comment (ivours) [inline CHANGELOG.md:15]: Should be "detailed" instead of "detail"

**Review by ivours — APPROVED** [review]

**Pull request merged**